### PR TITLE
Support multiple failure files per package

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**

### CODE REVIEW

File: protobuf-plugin/plugin/src/main/groovy/org/spine3/gradle/protobuf/failures/FailuresGenPlugin.groovy [hunk 1]
```diff
@@ -142,15 +151,42 @@ class FailuresGenPlugin implements Plugin<Project> {
...
     }
 
+    private static String getJavaOuterClassName(FileDescriptorProto descriptor) {
+        String outerClassNameFromOptions = descriptor.options.javaOuterClassname
+        if (!outerClassNameFromOptions.isEmpty()) {
+            return outerClassNameFromOptions;
+        }
+
+        final String fullFileName = descriptor.getName()
+        int lastBackslash = fullFileName.lastIndexOf('/')
```
Comment (armiol) [inline protobuf-plugin/plugin/src/main/groovy/org/spine3/gradle/protobuf/failures/FailuresGenPlugin.groovy]: Is it `final`?

Comment (alexander-yevsyukov) [inline protobuf-plugin/plugin/src/main/groovy/org/spine3/gradle/protobuf/failures/FailuresGenPlugin.groovy]: Fixed.

**Comment by alexander-yevsyukov** [comment]
So far we don't have any tests in the tools. I do think we need tests. The problem is that it's hardly testable in the way it's now: you cannot use the version of the code you use on the test data because it must be published and _then_ added to the build script. Otherwise it won't build.

**Comment by armiol** [comment]
@alexander-yevsyukov as discussed, let's test the parsing utilities only.

**Comment by alexander-yevsyukov** [comment]
Issue added: https://github.com/SpineEventEngine/tools/issues/33

**Comment by alexander-yevsyukov** [comment]
@armiol I've refactored and added some testing. 

More work on making this code smarter needs to be done, but as another PR. Here's the corresponding issue: #34

**Comment by armiol** [comment]
@alexander-yevsyukov LGTM.

**Pull request merged**